src/main/java/com/google/sps/algorithms/MatchQuery.java

```diff
@@ -2,19 +2,18 @@
 
 import com.google.sps.data.MatchRequest;
 import com.google.sps.data.User;
+import com.google.sps.data.NonPersistentUserRepository;
 import java.util.ArrayList;
 import java.util.Collection;
 
 public final class MatchQuery {
 
```

Comment:
Rather than depending on the NonPersistentUserRepository implementation, see if you can depend on the UserRepository interface instead. This allows the MatchQuery class can focus on the logic for matching users without having to know about the UserRepository implementation.

One way to achieve this is to create the NonPersistentUserRepository and call addFakeMentors() outside of the MatchQuery class. You would then pass a UserRepository object into the MatchQuery constructor and set it as a member variable (this is called dependency injection: https://www.vogella.com/tutorials/DependencyInjection/article.html).

Reply:
I tried to do this, but couldn't get it working correctly. Maybe we can talk about it in the sync today.

Reply:
Discussed in sync